### PR TITLE
treewide: use optionalString instead of 'then ""'

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/cgminer.nix [hunk 1]
```diff
@@ -11,7 +11,7 @@ let
     mapAttrsToList (n: v: ''"${n}": "${(concatStringsSep "," (map convType v))}"'')
       (foldAttrs (n: a: [n] ++ a) [] cfg.hardware);
   mergedConfig = with builtins;
-    mapAttrsToList (n: v: ''"${n}":  ${if isBool v then "" else ''"''}${convType v}${if isBool v then "" else ''"''}'')
+    mapAttrsToList (n: v: ''"${n}":  ${if isBool v then (convType v) else ''"${convType v}"''}'')
```
Comment (roberth) [inline nixos/modules/services/misc/cgminer.nix:14]:
```suggestion
    mapAttrsToList (n: v: ''"${n}":  ${if isBool v then convType v else ''"${convType v}"''}'')
```

File: nixos/modules/services/continuous-integration/jenkins/default.nix
Comment (roberth) [inline nixos/modules/services/continuous-integration/jenkins/default.nix:1]: :heavy_check_mark:

File: nixos/modules/services/backup/borgbackup.nix
Comment (roberth) [inline nixos/modules/services/backup/borgbackup.nix:1]: :heavy_check_mark:

File: nixos/modules/services/misc/sourcehut/default.nix
Comment (roberth) [inline nixos/modules/services/misc/sourcehut/default.nix:1]: :heavy_check_mark:

File: nixos/modules/services/monitoring/prometheus/exporters/pve.nix
Comment (roberth) [inline nixos/modules/services/monitoring/prometheus/exporters/pve.nix:1]: :heavy_check_mark:

File: nixos/modules/services/networking/biboumi.nix
Comment (roberth) [inline nixos/modules/services/networking/biboumi.nix:1]: :heavy_check_mark:

File: nixos/modules/services/networking/cjdns.nix
Comment (roberth) [inline nixos/modules/services/networking/cjdns.nix:1]: :heavy_check_mark:

File: nixos/modules/services/networking/libreswan.nix [hunk 2]
```diff
@@ -14,8 +14,8 @@ let
     nonchars = filter (x : !(elem x.value chars))
                (imap0 (i: v: {ind = i; value = v;}) (stringToCharacters str));
   in
-    if length nonchars == 0 then ""
-    else substring (head nonchars).ind (add 1 (sub (last nonchars).ind (head nonchars).ind)) str;
+    lib.optionalString (length nonchars != 0)
```
Comment (roberth) [inline nixos/modules/services/networking/libreswan.nix:17]:
```suggestion
    lib.optionalString (nonchars != [ ])
```

**Review by roberth — COMMENTED** [review]
Checked up to and including libreswan.

File: nixos/modules/services/misc/cgminer.nix (hunk 1 above)
Comment (Stunkymonkey) [reply nixos/modules/services/misc/cgminer.nix:14]: i am applying your suggestions. and will squash before merging.

**Review by ghost — APPROVED** [review]

**Comment by ehmry** [comment]
If the rebuilding and non-rebuilding changes could be sperated then merging non-builds would hardly need review.

File: nixos/modules/services/networking/murmur.nix [hunk 3]
```diff
@@ -32,17 +32,17 @@ let
     bonjour=${boolToString cfg.bonjour}
     sendversion=${boolToString cfg.sendVersion}
 
-    ${if cfg.registerName     == "" then "" else "registerName="+cfg.registerName}
-    ${if cfg.registerPassword == "" then "" else "registerPassword="+cfg.registerPassword}
-    ${if cfg.registerUrl      == "" then "" else "registerUrl="+cfg.registerUrl}
-    ${if cfg.registerHostname == "" then "" else "registerHostname="+cfg.registerHostname}
+    ${optionalString (cfg.registerName != "") "registerName=${cfg.registerName}"}
+    ${optionalString (cfg.registerPassword == "") "registerPassword=${cfg.registerPassword}"}
```
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/services/networking/murmur.nix:36]: sorry.

please look at #295407

[the remark this answers is not in the document]
Comment (infinisil) [reply nixos/modules/services/networking/murmur.nix:36]: No worries, thanks for the quick fix!

**Pull request merged**